### ISSUE TITLE
no difference when using N=0 and N=1 on bowtie2 2.2.6

### ISSUE THREAD

**Issue by linjc2011** [BenLangmead/bowtie2#94]
Hi.
When I used bowtie2 2.2.6 on Ubuntu 16.04 LTS for mapping arabidopsis reads. The mapping rates have no difference when I set N=0 or N=1.
However, when I used bowtie 2.1.0 on Ubuntu 14.04 LTS for mapping the same fastq file, the mapping rates are different, which N=1 can produced 10% higher than N=0.
Do anybody meet with this before?
Thanks.

**Comment by BenLangmead** [comment]
I see a difference:

```
13:00:29-langmead~/software/bowtie2-2.2.6$ ./bowtie2 -x example/index/lambda_virus -U example/reads/reads_1.fq > /dev/null -t -N 0
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:01
10000 reads; of these:
  10000 (100.00%) were unpaired; of these:
    596 (5.96%) aligned 0 times
    9404 (94.04%) aligned exactly 1 time
    0 (0.00%) aligned >1 times
94.04% overall alignment rate
Time searching: 00:00:01
Overall time: 00:00:01
13:00:32-langmead~/software/bowtie2-2.2.6$ ./bowtie2 -x example/index/lambda_virus -U example/reads/reads_1.fq > /dev/null -t -N 1
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:01
10000 reads; of these:
  10000 (100.00%) were unpaired; of these:
    342 (3.42%) aligned 0 times
    9658 (96.58%) aligned exactly 1 time
    0 (0.00%) aligned >1 times
96.58% overall alignment rate
Time searching: 00:00:01
Overall time: 00:00:01
```

I see this for v2.2.6, v2.2.9 and v2.3.1.  Could you share an example where there's no difference between `-N 0` and `-N 1`?

**Comment by BenLangmead** [comment]
See issue #104 for progress

**Comment by ch4rr0** [comment]
A fix for this has been released as of v2.3.3.
